acme/acme/standalone.py

```diff
@@ -43,7 +43,14 @@ def __init__(self, *args, **kwargs):
 
     def _wrap_sock(self):
         self.socket = crypto_util.SSLSocket(
-            self.socket, certs=self.certs, method=self.method)
+            self.socket, {}, cert_selection=self.cert_selection,
+            alpn_selection=getattr(self, 'alpn_selection', None),
+            method=self.method)
+
+    def cert_selection(self, connection):
+        """Callback selecting certificate for connection."""
+        server_name = connection.get_servername()
+        return self.certs.get(server_name, None)
```

Comment:
Let's prefix `cert_selection` and `alpn_selection` with `_` to indicate they're for internal use only?

Reply:
I don't think they should be for internal use only.

Reply:
how do you mean? the selection routines can be overriden by subclasses, but what use might someone have for consuming them in the API for `TLSServer`?

Reply:
Oh, right, they definitely should be private. I was reading this on a phone and misread the context.